contracts/core/Dispatcher.sol

```diff
@@ -453,25 +481,33 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
         emit SendPacket(sender, channelId, packet, sequence, timeoutTimestamp);
     }
 
-    function _verifyConnectIbcChannelProof(
+    function _connectChannel(
+        IbcChannelReceiver portAddress,
         CounterParty calldata local,
         string[] calldata connectionHops,
         ChannelOrder ordering,
-        bool isChanConfirm,
-        CounterParty calldata counterparty,
-        Ics23Proof calldata proof
+        bool feeEnabled,
+        CounterParty calldata counterparty
     ) internal {
-        consensusStateManager.verifyMembership(
-            proof,
-            channelProofKey(local.portId, local.channelId),
-            channelProofValue(
-                isChanConfirm ? ChannelState.CONFIRM_PENDING : ChannelState.ACK_PENDING,
-                ordering,
-                local.version,
-                connectionHops,
-                counterparty
-            )
+        portAddress.onChannelConnect(local.channelId, counterparty.version);
+
```

Comment:
Consider moving the `onChannelConnect` call to the end of this function so as to follow the [checks-effects-interactions](https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html) pattern. 

I don't think this particularly insecure code if someone _were_ to reenter this contract, but in general it is always good to follow this pattern to prevent random code injections that can come from recursively calling this function in the `onChannelConnect` handler